tests/unit/rpc/RPCHelpersTests.cpp

```diff
@@ -547,6 +556,264 @@ TEST_F(RPCHelpersTest, ParseIssue)
     );
 }
 
+TEST_F(RPCHelpersTest, AccountHoldsFixLPTAmendmentDisabled)
+{
+    auto ammAccount = getAccountIdWithString(kAMM_ACCOUNT);
+    auto account = getAccountIdWithString(kACCOUNT);
+
+    auto const lptRippleState = createRippleStateLedgerObject(
+        kLPTOKEN_CURRENCY, kAMM_ACCOUNT, 100, kACCOUNT, 100, kAMM_ACCOUNT, 100, kTXN_ID, 3
+    );
+    auto const lptRippleStateKk = ripple::keylet::line(ammAccount, account, ripple::to_currency(kLPTOKEN_CURRENCY)).key;
+
+    // trustline fetched twice. once in accountHolds and once in isFrozen
+    EXPECT_CALL(*backend_, doFetchLedgerObject(lptRippleStateKk, testing::_, testing::_)).Times(2);
+    ON_CALL(*backend_, doFetchLedgerObject(lptRippleStateKk, testing::_, testing::_))
+        .WillByDefault(testing::Return(lptRippleState.getSerializer().peekData()));
+
+    auto const ammID = ripple::uint256{54321};
+    auto const ammAccountKk = ripple::keylet::account(ammAccount).key;
+    auto const ammAccountRoot = createAccountRootObject(kAMM_ACCOUNT, 0, 2, 200, 2, kINDEX1, 2, 0, ammID);
+
+    EXPECT_CALL(*backend_, doFetchLedgerObject(ammAccountKk, testing::_, testing::_)).Times(1);
+    ON_CALL(*backend_, doFetchLedgerObject(ammAccountKk, testing::_, testing::_))
+        .WillByDefault(Return(ammAccountRoot.getSerializer().peekData()));
+
+    EXPECT_CALL(*mockAmendmentCenterPtr_, isEnabled(testing::_, Amendments::fixFrozenLPTokenTransfer, testing::_))
+        .Times(1);
+    ON_CALL(*mockAmendmentCenterPtr_, isEnabled(testing::_, Amendments::fixFrozenLPTokenTransfer, testing::_))
+        .WillByDefault(testing::Return(false));
+
+    boost::asio::spawn(ctx_, [&, this](boost::asio::yield_context yield) {
+        auto ret = accountHolds(
+            *backend_,
+            *mockAmendmentCenterPtr_,
+            0,
+            account,
+            ripple::to_currency(kLPTOKEN_CURRENCY),
+            ammAccount,
+            true,
+            yield
+        );
+        EXPECT_EQ(ret.mantissa(), 1000000000000000);
+    });
+    ctx_.run();
+}
+
+TEST_F(RPCHelpersTest, AccountHoldsLPTokenNotAMMAccount)
+{
+    auto account = getAccountIdWithString(kACCOUNT);
+    auto account2 = getAccountIdWithString(kACCOUNT2);
+
+    auto const usdRippleState =
+        createRippleStateLedgerObject("USD", kACCOUNT2, 100, kACCOUNT, 100, kACCOUNT2, 100, kTXN_ID, 3);
+    auto const usdRippleStateKk = ripple::keylet::line(account2, account, ripple::to_currency("USD")).key;
+
+    // trustline fetched twice. once in accountHolds and once in isFrozen
+    EXPECT_CALL(*backend_, doFetchLedgerObject(usdRippleStateKk, testing::_, testing::_)).Times(2);
+    ON_CALL(*backend_, doFetchLedgerObject(usdRippleStateKk, testing::_, testing::_))
```

Comment:
All these ON_CALL should be instead setup in the EXPECT_CALL with WillOnce or WillRepeatedly

Reply:
fixed